web-common/src/features/sources/sourceUtils.ts

```diff
@@ -45,3 +53,56 @@ export function inferSourceName(connector: V1Connector, path: string) {
 
   return sanitizeEntityName(fileName);
 }
+
+function getFileTypeFromName(fileName) {
+  const fileType = fileName.split(".").pop();
+
+  if (!fileType) return "";
+
+  if (fileType === "gz") {
+    return fileName.split(".").slice(-2).shift();
+  }
+
+  return fileType;
+}
+
+export function sourceErrorTelemetryHandler(
```

Comment:
Would something like `emitSourceErrorTelemetry` make more sense here? Same for the next method.